src/iris/api.py

```diff
@@ -831,6 +831,41 @@ def gen_where_filter_clause(connection, filters, filter_types, kwargs):
     return where
 
 
+def generate_grouped_query(query, columns, fields):
+    '''modify query to retrieve total counts for distinct combinations of specified columns while avoiding the return of each unique row individually'''
+    # avoid DISTINCT grouping
+    modified_query = query.replace("SELECT DISTINCT", "SELECT")
+    # Construct the counts query
+    group_by_clause = ", ".join([f"`{column}`" for column in fields if column in columns])
```

Comment:
Can we add validation that at least one valid field was provided and 4xx otherwise, to avoid a broken query?

Reply:
added